x/staking/src/keeper/query.rs

```diff
@@ -63,11 +101,80 @@ impl<
             amount: tokens.to_uint_floor(),
         };
         Ok(DelegationResponse {
-            delegation,
-            balance,
+            delegation: Some(delegation),
+            balance: Some(balance),
         })
     }
 
+    pub fn query_delegator_delegations<DB: Database>(
+        &self,
+        ctx: &QueryContext<DB, SK>,
+        query: QueryDelegatorDelegationsRequest,
+    ) -> QueryDelegatorDelegationsResponse {
+        let store = ctx.kv_store(&self.store_key);
+        let store = store.prefix_store(DELEGATION_KEY);
+        let key = query.delegator_addr.prefix_len_bytes();
+        let (p_result, iterator) = store
+            .into_range(..)
+            .maybe_paginate(query.pagination.map(gears::ext::Pagination::from));
+
+        let delegation_responses = iterator
+            .filter_map(|(k, bytes)| {
+                if k.starts_with(&key) {
+                    Delegation::decode_vec(&bytes).ok()
```

Comment:
Is it okay to ignore invalid keys?

Reply:
I'm not sure that it is better than an error. I'll update.